pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -119,6 +119,37 @@ func (self *RefsHelper) ResetToRef(ref string, strength string, envVars []string
 	return nil
 }
 
+func (self *RefsHelper) CreateSortOrderMenu(onSelected func(sortOrder string) error) error {
+	type sortOrderWithKey struct {
+		key       types.Key
+		label     string
+		sortKey   string
+		sortOrder string
+	}
+	sortKeys := []sortOrderWithKey{
+		{label: self.c.Tr.SortAlphabetical, sortKey: "refname", sortOrder: "alphabetical", key: 'a'},
+		{label: self.c.Tr.SortByDate, sortKey: "-committerdate", sortOrder: "date", key: 'd'},
+	}
+
+	menuItems := lo.Map(sortKeys, func(row sortOrderWithKey, _ int) *types.MenuItem {
+		return &types.MenuItem{
+			LabelColumns: []string{
+				row.label,
+				style.FgDefault.Sprintf("--sort=%s", row.sortKey),
```

Comment:
I noticed one more thing: `FgDefault` is redundant, this might as well be `fmt.Sprintf(...)`. However, this made me think that maybe we want to use a different color? I looked a bit through other menus that have columns, and we are being quite inconsistent with what colors we use in cases like this. The "reset" menu and the "discard changes" menu both use red, but that's probably because those are destructive commands. The "bulk submodule options" menu uses a traffic light, but that's also a bit of a special case. I suppose FgYellow might work well as a middle ground, what do you think?

And I have another question while I'm here. I'm currently working on a followup PR to add sorting to local branches too. In that case we have a third menu entry, which is "recency" (today's behavior). There's no sort key for this, what it does is sort by `-committerdate` and then use some complicated algorithm to sort branches based on reflog entries somehow. I wonder what to put in the second label column for this. We could show something like
```
r Recency       (based on reflog)
d Date          --sort=-committerdate
a Alphabetical  --sort=refname
```
Any thoughts?

Reply:
I initially used the FgRed but then decided that it's not appropriate and just went with the default. If there's no convention for colors in the menu we could adopt yellow.

Regarding the sorting for local branches, I've already implemented this on a separate branch, just need to add a test case. I was going to PR it after this one gets merged. I'm not sure how far in you are, but please see the "local-branch-sort-menu" in my forked repo. I moved out the "sortOrderWithKey" as a separate Options type and that gets passed along with the "onPressed" callback. So each callee can define sort options appropriate for their context.

Reply:
Specifically: https://github.com/hosaka/lazygit/commit/823faafeffc1a74abea886f1d772681181c7ba0a

Reply:
Ah nice, I hadn't expected you to work on this so quickly. Well, I'm basically done except for the test; my commit looks quite similar to yours, but I did some details a little differently. Here it is: 268c56bd3e20

Reply:
Very similar indeed! I chose to add a CommitterDate prop to the branch struct, as to avoid storing non-recency data in the Recency :) Also I think a standalone "SortOrdedOptions" is a bit better than a string array, allowing local/remote branches (maybe tags at some point?) to define their own contents for the sort menu, at the cost of some duplication of course.

Reply:
> I chose to add a CommitterDate prop to the branch struct, as to avoid storing non-recency data in the Recency

Interesting, for me it's still the same semantics (how "recent" is this branch), just with a slightly different meaning for what this means exactly (how recently was this branch last checked out, vs. how recently was the last commit on this branch made.) I find CommitterDate misleading because it sounds like it would store a string in date format; so it would have to be called CommitterRecency or something like that. But I'm not sure it's really worth making the distinction.

Also, one problem with your approach is that the last few branches (the ones that don't have reflog recency information for some reason) used to have an empty recency column, but in your branch they show the commiterdate recency, which is confusing.

> Also I think a standalone "SortOrdedOptions" is a bit better than a string array

I found it more important to avoid the code duplication, but I could live with either solution.

Reply:
Yeah, CommitterDate is not all necessarry, you make a good point!

On the SortOrderOptions, duplication can be easily solved with a set of "default" sort options, which callees can then extend as necessary. I think having a new type is appropriate when more than 1 or 2 places need to pass custom options. But I'm also not too religious about it.

Should we perhaps carry this discussion in it's own PR when this gets merged?

Reply:
> On the SortOrderOptions, duplication can be easily solved with a set of "default" sort options

Except that this makes it hard to have them in a different order. Maybe I'm too picky here, but I like to have "alphabetical,date" for remote branches but "recency,date,alphabetical" for local branches, with the rationale that the default option should be first, and then the remaining options in the order of likelihood of being chosen by users. That's the reason why I defined the options in the function as a map, and let callers specify just the order in which they appear as a slice of strings. (I'm open to change my mind about this, it could also be seen as valuable to have the options appear in the same order in every menu.)

> Should we perhaps carry this discussion in it's own PR when this gets merged?

Absolutely. The only thing missing for this PR to be merged is the yellow color for the second menu column.

Reply:
Fixed and pushed.